src/test/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCaseFlagTest.java

```diff
@@ -50,4 +68,124 @@ void shouldNotConfigureCaseFlagIfFeatureFlagFalse() {
             .doesNotContain(CASEWORKER_CASE_FLAG);
     }
 
+    @Test
+    void shouldSuccessfullyAddFlagRepresentative() {
+        //Given
+        final CaseData caseData = caseData();
+        caseData.setNote("This is a test note");
+
+        final CicCase cicCase = new CicCase();
+        final Set<RepresentativeCIC> set = new HashSet<>();
+        set.add(RepresentativeCIC.REPRESENTATIVE);
+        cicCase.setRepresentativeCIC(set);
+        cicCase.setRepresentativeFullName("Jane Doe");
+        cicCase.setNotifyPartyRepresentative(set);
+        caseData.setCicCase(cicCase);
+        final CaseDetails<CaseData, State> updatedCaseDetails = new CaseDetails<>();
+        final CaseDetails<CaseData, State> beforeDetails = new CaseDetails<>();
+        updatedCaseDetails.setData(caseData);
+        updatedCaseDetails.setId(TEST_CASE_ID);
+        updatedCaseDetails.setCreatedDate(LOCAL_DATE_TIME);
+
+        //When
+        AboutToStartOrSubmitResponse<CaseData, State> response =
+            caseworkerCaseFlag.aboutToSubmit(updatedCaseDetails, beforeDetails);
+        SubmittedCallbackResponse submittedCallbackResponse = caseworkerCaseFlag.flagCreated(updatedCaseDetails, beforeDetails);
+
+
+        //Then
+        assertThat(submittedCallbackResponse).isNotNull();
```

Comment:
More assertions can be done apart from just not null validation. (there are many test cases need to be updated)

Reply:
@sjamihmcts ^^^

Reply:
Updated the tests with more assertions